hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java

```diff
@@ -470,4 +461,17 @@ private void closeDbStore() {
     LOG.info("SchemaV3 db is stopped at {} for volume {}", containerDBPath,
         getStorageID());
   }
+
+  public void compactDb() {
+    File dbFile = new File(getDbParentDir(), CONTAINER_DB_NAME);
+    String dbFilePath = dbFile.getAbsolutePath();
+    try {
+      // Calculate number of files per level and size per level
+      RawDB rawDB =
+          DatanodeStoreCache.getInstance().getDB(dbFilePath, getConf());
+      rawDB.getStore().compactionIfNeeded();
```

Comment:
Are the metrics that get generated which we capture today for compaction duration? Also, it would be good to add volume labels for each metric.

Reply:
Thanks @kerneltime for the review, goog idea.